ksql-examples/src/main/java/io/confluent/ksql/datagen/DataGenProducer.java

```diff
@@ -73,74 +82,19 @@ public void populateTopic(
     SessionManager sessionManager = new SessionManager();
 
     for (int i = 0; i < messageCount; i++) {
-      Object generatedObject = generator.generate();
 
-      if (!(generatedObject instanceof GenericRecord)) {
-        throw new RuntimeException(String.format(
-            "Expected Avro Random Generator to return instance of GenericRecord, found %s instead",
-            generatedObject.getClass().getName()
-        ));
-      }
-      GenericRecord randomAvroMessage = (GenericRecord) generatedObject;
-
-      List<Object> genericRowValues = new ArrayList<>();
-
-      SimpleDateFormat timeformatter = null;
-
-      /**
-       * Populate the record entries
-       */
-      String sessionisationValue = null;
-      for (Schema.Field field : avroSchema.getFields()) {
-
-        boolean isSession = field.schema().getProp("session") != null;
-        boolean isSessionSiblingIntHash =
-            field.schema().getProp("session-sibling-int-hash") != null;
-        String timeFormatFromLong = field.schema().getProp("format_as_time");
-
-        if (isSession) {
-          String currentValue = (String) randomAvroMessage.get(field.name());
-          String newCurrentValue = handleSessionisationOfValue(sessionManager, currentValue);
-          sessionisationValue = newCurrentValue;
-
-          genericRowValues.add(newCurrentValue);
-        } else if (isSessionSiblingIntHash && sessionisationValue != null) {
-
-          // super cheeky hack to link int-ids to session-values - if anything fails then we use
-          // the 'avro-gen' randomised version
-          handleSessionSiblingField(
-              randomAvroMessage,
-              genericRowValues,
-              sessionisationValue,
-              field
-          );
-
-        } else if (timeFormatFromLong != null) {
-          Date date = new Date(System.currentTimeMillis());
-          if (timeFormatFromLong.equals("unix_long")) {
-            genericRowValues.add(date.getTime());
-          } else {
-            if (timeformatter == null) {
-              timeformatter = new SimpleDateFormat(timeFormatFromLong);
-            }
-            genericRowValues.add(timeformatter.format(date));
-          }
-        } else {
-          genericRowValues.add(randomAvroMessage.get(field.name()));
-        }
-      }
-
-      GenericRow genericRow = new GenericRow(genericRowValues);
+      final Pair<String, GenericRow> genericRowPair = generateOneGenericRow(
```

Comment:
nit: imo it would be cleaner to just have this return a ProducerRecord.

Reply:
I see the point but generating `GenericRow` will keep the door open to use this method to write to stores other than Kafka such as file, etc.